tests/tests/force_settle_fee_tests.cpp

```diff
@@ -0,0 +1,1105 @@
+
+         asset_create_operation creator;
+         creator.issuer = issuer;
+         creator.fee = asset();
+         creator.symbol = name;
+         creator.precision = precision;
+
+         creator.common_options.max_supply = max_supply;
+         creator.common_options.market_fee_percent = market_fee_percent;
+         if (issuer == GRAPHENE_WITNESS_ACCOUNT)
+            flags |= witness_fed_asset;
+         creator.common_options.issuer_permissions = flags;
+         creator.common_options.flags = flags & ~global_settle;
+         creator.common_options.core_exchange_rate = price(asset(1, asset_id_type(1)), asset(1));
+
+         creator.bitasset_opts = bitasset_options();
+         creator.bitasset_opts->force_settlement_offset_percent = force_settlement_offset_percent;
+         creator.bitasset_opts->short_backing_asset = backing_asset;
+         if (force_settlement_fee_percent.valid()) {
+            creator.bitasset_opts->extensions.value.force_settle_fee_percent = force_settlement_fee_percent;
+         }
+
+         trx.operations.push_back(std::move(creator));
+         trx.validate();
+         processed_transaction ptx = PUSH_TX(db, trx, ~0);
+         trx.operations.clear();
+         return db.get<asset_object>(ptx.operation_results[0].get<object_id_type>());
+      } FC_CAPTURE_AND_RETHROW((name)(issuer))
+   }
+};
+
+
+/**
+ * Test the effects of the new force settlement fee from BSIP87
+ */
+BOOST_FIXTURE_TEST_SUITE(force_settle_tests, force_settle_database_fixture)
+
+   /**
+    * Test when one holder of a smart asset force settles (FS) their holding when there are two debtors
+    *
+    * There are three primary actors: michael, paul, rachel
+    *
+    * 1. Asset owner creates the smart coin called bitUSD
+    * 2. The feed price is 20 satoshi bitUSD for 1 satoshi Core -> 0.2 bitUSD for 0.00001 Core = 20000 bitUSD for 1 Core
+    * 3. Michael borrows 0.06 bitUSD (6 satoshis of bitUSD) from the blockchain with a high amount of collateral
+    * 4. Paul borrows 1000 bitUSD (100000 satoshis of bitUSD) from the blockchain with a low amount of collateral
+    * 5. Paul gives Rachel 200 bitUSD
+    * 6. Rachel force-settles 20 bitUSD which should be collected from Paul's debt position
+    * because of its relatively lower collateral ratio
+    *
+    * The force-settlement by Rachel should account for both the force-settlement offset fee,
+    * and the new force settlement fee from BSIP87.
+    *
+    * Michael's debt and balances should be unaffected by the activities of Paul and Rachel
+    */
+   BOOST_AUTO_TEST_CASE(force_settle_fee_1_test) {
+      try {
+         ///////
+         // Initialize the scenario
+         ///////
+         // Get around Graphene issue #615 feed expiration bug
+         generate_blocks(HARDFORK_615_TIME);
+         generate_blocks(db.get_dynamic_global_properties().next_maintenance_time);
+
+         // Advance to the when the force-settlement fee activates
```

Comment:
Ideally we need test cases to test that operations with new extensions (as well as being nested in proposals) aren't allowed before the hard fork.

Reply:
Thank you @MichelSantos.
Only this hasn't been addressed. Commenting here, I meant the hardfork checks for `asset_create_operation` and `asset_update_bitasset_operation`.

Reply:
Update: there is one test case for asset creation around line 411. I didn't find others.